### PR TITLE
Adc calibration issue

### LINKED ISSUE #384: ADC Calibrated Reading introduces offset when Data rate is changed
After calibrating the ADC input at 38400 data rate, the measurements at 20SPS introduce some offset voltage around 3mV.
- both ADC1 and ADC2  reading exhibit the same offset, but ADC1 has worse offset
-

### PR DESCRIPTION
**Pull request by sjpark608**
Proper handling of code value conversion
- REFERENCE_VOLTAGE shouldn't be used to convert voltage
- REFERENCE shifting must be done in raw Code value
This should fix #384

### CODE REVIEW

File: src/edgepi/adc/adc_voltage.py [hunk 1]
```diff
@@ -69,9 +72,15 @@ def code_to_voltage(code: list[int], adc_info: ADCReadInfo, calibs: CalibParam)
...
-        code_val = code_val - 2**num_bits
+
+    if _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 4:
+        code_val = code_val - ADC1_UPPER_LIMIT
+    elif _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 3:
+        code_val = code_val - ADC2_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 4:
+        code_val = code_val + ADC1_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 3:
+        code_val = code_val + ADC2_UPPER_LIMIT
```
Comment (farzadpanahi) [inline src/edgepi/adc/adc_voltage.py:83]: what do `adc_info.num_data_bytes == 4` and  `adc_info.num_data_bytes == 3` mean? is that how we identify adc1 and adc2 ? should we add comments to clarify?

File: src/edgepi/adc/adc_voltage.py [hunk 2]
```diff
@@ -96,12 +105,17 @@ def code_to_voltage_single_ended(code: list[int], adc_info: ADCReadInfo, calibs:
...
-        code_val = code_val - 2**num_bits
+    
+    if _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 4:
+        code_val = code_val - ADC1_UPPER_LIMIT
+    elif _is_negative_voltage(code_bits) and adc_info.num_data_bytes == 3:
+        code_val = code_val - ADC2_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 4:
+        code_val = code_val + ADC1_UPPER_LIMIT
+    elif adc_info.num_data_bytes == 3:
+        code_val = code_val + ADC2_UPPER_LIMIT
```
Comment (farzadpanahi) [inline src/edgepi/adc/adc_voltage.py:116]: same here

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: src/edgepi/adc/adc_voltage.py (hunk 1 above)
Comment (sjpark608) [reply src/edgepi/adc/adc_voltage.py]: [0c5a3d0](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/385/commits/0c5a3d08ee9674de0c0c474223d095c230d743dc)

I created a constants for these numbers

File: src/edgepi/adc/adc_voltage.py (hunk 2 above)
Comment (sjpark608) [reply src/edgepi/adc/adc_voltage.py]: [0c5a3d0](https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/385/commits/0c5a3d08ee9674de0c0c474223d095c230d743dc)

**Review by farzadpanahi — APPROVED** [review]

**Pull request merged**